share/smack/frontend.py

```diff
@@ -313,7 +314,7 @@ def default_build_libs(args):
     if args.pthread:
         libs += ['pthread.c']
 
-    if args.strings or args.memory_safety or args.integer_overflow:
+    if args.strings:
         libs += ['string.c']
```

Comment:
Why were we enabling this for memory safety and overflow checking?

Reply:
Yeah, I was using git blame to try to figure it out, but no luck. I think it must be related to SVCOMP benchmarks.
Since I would like to completely rewrite our SVCOMP script, I figured I might as well clean this up too.